src/dashboard/apps/consent/views.py

```diff
@@ -2,92 +2,102 @@
 
 from django.contrib import messages
 from django.core.exceptions import PermissionDenied
-from django.db.models import TextField
+from django.db.models import Q, TextField
 from django.db.models.functions import Cast
-from django.shortcuts import get_object_or_404, redirect, render
+from django.shortcuts import get_object_or_404, redirect
 from django.urls import reverse_lazy as reverse
 from django.utils import timezone
 from django.utils.translation import gettext_lazy as _
-from django.views.generic import TemplateView
+from django.views.generic import FormView, TemplateView
 
 from apps.core.models import Entity
 
+from ..auth.models import DashboardUser
 from . import AWAITING, VALIDATED
+from .forms import ConsentForm
+from .mixins import BreadcrumbContextMixin
 from .models import Consent
 
 
-class IndexView(TemplateView):
+class IndexView(BreadcrumbContextMixin, TemplateView):
     """Index view of the consent app."""
 
     template_name = "consent/index.html"
+    breadcrumb_current = _("Consent")
 
-    def get_context_data(self, **kwargs):
-        """Add custom context to the view."""
+    def get_context_data(self, **kwargs):  # noqa: D102
         context = super().get_context_data(**kwargs)
-        context["entities"] = self.request.user.get_entities()
+        user: DashboardUser = self.request.user  # type: ignore
+        context["entities"] = user.get_entities()
 
-        context["breadcrumb_data"] = {
-            "current": _("Consent"),
-        }
         return context
 
 
-def consent_form_view(request, slug=None):
-    """Manage consent forms.
+class ConsentFormView(BreadcrumbContextMixin, FormView):
+    """Updates the status of consents."""
 
-    This function performs the following actions:
-    - Retrieves the entity associated with the given slug, if provided.
-    - Fetches consents awaiting validation for the current user and the specified
-    entity.
-    - If a POST request is received, updates the consent status to either VALIDATED
-    or AWAITING based on user selections and existing data.
-    """
     template_name = "consent/manage.html"
+    form_class = ConsentForm
+
+    breadcrumb_links = [
+        {"url": reverse("consent:index"), "title": _("Consent")},
+    ]
+    breadcrumb_current = _("Manage Consents")
+
+    def form_valid(self, form):  # noqa: D102
```

Comment:
So you don't want to write what you do :sweat_smile:

Reply:
I wouldn't write what I do without my lawyer present. :stuck_out_tongue: 

well indeed, I don't know what happened here :sweat_smile: